### PR TITLE
[tests] add traffic analysis for Cert_5_3_10 as TestPlan

### PR DESCRIPTION
**Pull request by zhaxia**

### CODE REVIEW

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 1]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        router1_id = self.nodes[ROUTER1].get_addr16()
+
+        self.nodes[ROUTER1].stop()
+        time.sleep(580)
+
+        # Send an ICMPv6 Echo Request from SED1 to ROUTER1 GUA 2003:: address
+        self.assertTrue(router1_addr is not None)
+        self.assertFalse(self.nodes[MED1].ping(router1_addr))
+
+        # Verify the ROUTER2 has removed all entries based on ROUTER1's Router ID
```
Comment (librasungirl) [inline tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: `router1_id` you are using is actually rloc16 address, not `ROUTER1's Router ID`
nit: ROUTER2 -> DUT_ROUTER2

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 2]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        self.assertTrue(self.nodes[MED1].ping(addr))
+
+        # Wait for sniffer got Address Notification messages
+        time.sleep(1)
+
+        # Verify DUT_ROUTER2 sent an Address Query Request
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/aq')
+        msg.assertSentToDestinationAddress(config.REALM_LOCAL_ALL_ROUTERS_ADDRESS)
+        command.check_address_query_request(msg)
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: I missed this function. Will update it.

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 3]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+
+        # Verify DUT_ROUTER2 received an Address Notification message
+        router1_messages = self.sniffer.get_messages_sent_by(ROUTER1)
+        msg = router1_messages.next_coap_message('0.02', '/a/an')
+        msg.assertSentToNode(self.nodes[DUT_ROUTER2])
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will remove it.

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 4]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        addr = self.nodes[MED1].get_addr("2003::/64")
+        self.assertTrue(addr is not None)
+        self.assertTrue(self.nodes[BR].ping(addr))
+
+        # Wait for sniffer got Address Notification messages
+        time.sleep(1)
+
+        # Verify DUT_ROUTER2 sent an Address Notification message
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/an')
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 5]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+
+        # Verify DUT_ROUTER2 didn't generate an Address Query Request
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        dut_router2_messages_temp = copy.deepcopy(dut_router2_messages)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/aq', False)
+        assert msg is None, "Error: The DUT_ROUTER2 sent an unexpected Address Query Request"
+
+        # Verify DUT_ROUTER2 forwarded ICMPv6 Echo Reply to MED1
+        msg = dut_router2_messages_temp.get_icmp_message(ipv6.ICMP_ECHO_RESPONSE)
+        assert msg is not None, "Error: The DUT_ROUTER2 didn's forward ICMPv6 Echo Reply to MED1"
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 6]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        msg.assertSentToNode(self.nodes[MED1])
+
+        # 6 DUT_ROUTER2: Power off ROUTER1 and wait 580 seconds to allow the LEADER to expire its Router ID
+        self.nodes[MED1].get_addrs()
+        router1_addr = self.nodes[ROUTER1].get_addr("2003::/64")
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 7]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        router1_id = self.nodes[ROUTER1].get_addr16()
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 8]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+
+        self.nodes[ROUTER1].stop()
+        time.sleep(580)
+
+        # Send an ICMPv6 Echo Request from SED1 to ROUTER1 GUA 2003:: address
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py (hunk 1 above)
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 9]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        self.assertTrue(router1_addr is not None)
+        self.assertFalse(self.nodes[MED1].ping(router1_addr))
+
+        # Verify the ROUTER2 has removed all entries based on ROUTER1's Router ID
+        eidcaches = self.nodes[DUT_ROUTER2].get_eidcaches()
+        self.assertFalse(any(router1_id == int(rloc, 16) for (_, rloc) in eidcaches))
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Good suggestion. I will add a method to process this case.

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 10]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+
+        # Verify DUT_ROUTER2 sent an Address Query Request
+        dut_router2_messages = self.sniffer.get_messages_sent_by(DUT_ROUTER2)
+        msg = dut_router2_messages.next_coap_message('0.02', '/a/aq')
+        msg.assertSentToDestinationAddress(config.REALM_LOCAL_ALL_ROUTERS_ADDRESS)
+
+        # 7 MED1: Power off MED1 and wait to allow DUT_ROUTER2 to timeout the child
+        med1_addr = self.nodes[MED1].get_addr("2003::/64")
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py [hunk 11]
```diff
@@ -91,52 +105,125 @@ def test(self):
...
+        self.assertTrue(med1_addr is not None)
+        self.nodes[MED1].stop()
+        time.sleep(config.MLE_END_DEVICE_TIMEOUT)
+
+        # BR sends two ICMPv6 Echo Request to MED1 GUA 2003:: address
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/Cert_5_3_10_AddressQuery.py]: Will update it

File: tests/scripts/thread-cert/config.py [hunk 12]
```diff
@@ -65,6 +65,7 @@
 INFINITE_COST_TIMEOUT = 90
 
 MAX_ADVERTISEMENT_INTERVAL = 32
+MLE_END_DEVICE_TIMEOUT = 240
```
[the remark this answers is not in the document]
Comment (zhaxia) [reply tests/scripts/thread-cert/config.py]: Will update it

**Pull request merged**